apache-composer error handling reports weird error

this source:

```javascript
const composer = require('openwhisk-composer')

module.exports = composer.sequence(composer.action('A', { filename: 'doesnotexist.js' }))
```

should give an error something like this:
```
{ Error: ENOENT: no such file or directory, open 'doesnotexist.js'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at Object.action (/Users/nickm/git/whisk/nickm/kui/node_modules/openwhisk-composer/composer.js:350:17)
    at Object.<anonymous> (/private/tmp/foo.js:3:45)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'doesnotexist.js',
  statusCode: 422 }
```

instead, `preview foo.js`, kui shows "Error: Cannot find module 'openwhisk-composer'", unless you do `preview @demos/if.js` *first*, in which case, the expected error is shown. weird

--- a/plugins/plugin-apache-composer/src/initRequirePath.ts
+++ b/plugins/plugin-apache-composer/src/initRequirePath.ts
@@ -1,0 +1,32 @@
+/*
+ * Copyright 2019 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as path from 'path'
+import { inBrowser } from '@kui-shell/core/core/capabilities'
+import * as Debug from 'debug'
+const debug = Debug('plugins/apache-composer/initRequirePath')
+
+// help compositions find our openwhisk-composer module
+export default async () => {
+  if (!inBrowser()) {
+    debug('adding node_modules to the require module path')
+    const appModulePath = await import('app-module-path')
+
+    // add the directory that encloses `openwhisk-composer`
+    // this is needed e.g. for `compose foo`
+    const root = path.dirname(require.resolve('openwhisk-composer/package.json'))
+    appModulePath.addPath(path.join(root, '..'))
+  }
+}

--- a/plugins/plugin-apache-composer/src/lib/utility/compile.ts
+++ b/plugins/plugin-apache-composer/src/lib/utility/compile.ts
@@ -33,19 +33,6 @@ import { create } from './usage'
 import { deployAction } from '../controller/client'
 import * as messages from './messages'
 
-// help compositions find our openwhisk-composer module
-export const initRequirePath = async () => {
-  if (!inBrowser()) {
-    debug('adding node_modules to the require module path')
-    const appModulePath = await import('app-module-path')
-
-    // add the directory that encloses `openwhisk-composer`
-    // this is needed e.g. for `compose foo`
-    const root = path.dirname(require.resolve('openwhisk-composer/package.json'))
-    appModulePath.addPath(path.join(root, '..'))
-  }
-}
-
 export const sourceToComposition = ({ inputFile, name = '', recursive = false }) => new Promise(async (resolve, reject) => {
   debug('validating source file', inputFile)
   const extension = inputFile.substring(inputFile.lastIndexOf('.') + 1)

--- a/plugins/plugin-apache-composer/src/plugin.ts
+++ b/plugins/plugin-apache-composer/src/plugin.ts
@@ -27,12 +27,9 @@ import appList from './lib/controller/cmd/app-list'
 import appConfig from './lib/controller/cmd/app-config'
 import editorBits from './lib/controller/cmd/editor-extensions'
 import * as usage from './usage'
-import { initRequirePath } from './lib/utility/compile'
 
 export default async (commandTree, prequire) => {
   debug('initializing')
-
-  await initRequirePath()
 
   commandTree.subtree('/composer', { usage: usage.composer })
 

--- a/plugins/plugin-apache-composer/src/preload.ts
+++ b/plugins/plugin-apache-composer/src/preload.ts
@@ -18,6 +18,7 @@ import { isHeadless } from '@kui-shell/core/core/capabilities'
 import { addPath } from '@kui-shell/core/core/find-file'
 import * as repl from '@kui-shell/core/core/repl'
 import { PluginRequire, PreloadRegistration } from '@kui-shell/core/models/plugin'
+import initRequirePath from './initRequirePath'
 import * as Debug from 'debug'
 const debug = Debug('plugins/composer/preload')
 
@@ -50,6 +51,9 @@ const listenForDrops = () => {
 const registration: PreloadRegistration = async (commandTree, prequire: PluginRequire) => {
   // listen for drag and drop
   listenForDrops()
+
+  // help compositions find our openwhisk-composer module
+  await initRequirePath()
 
   // give visibility to our @demos directory on the module path
   addPath(dirname(require.resolve('@kui-shell/plugin-apache-composer/lib/@demos/hello.js')))

--- a/plugins/plugin-apache-composer/src/test/composer1/composer-viz.ts
+++ b/plugins/plugin-apache-composer/src/test/composer1/composer-viz.ts
@@ -51,6 +51,7 @@ const requireAbsolute = composerInput('require-absolute.js')
 const requireRelative = composerInput('require-relative.js')
 const fsRead = composerInput('fs-read.js')
 const addSubscription = composerErrorInput('addSubscription.js')
+const owComposerErr = composerErrorInput('openwhisk-composer-throw-err.js')
 
 /**
  * Here starts the test
@@ -61,6 +62,10 @@ describe('show the composer visualization without creating openwhisk assets', fu
   after(common.after(this))
 
   it('should have an active repl', () => cli.waitForRepl(this.app))
+
+  it('should show error thrown by openwhisk-composer node_module', () => cli.do(`preview ${owComposerErr.path}`, this.app)
+    .then(cli.expectError(0, `no such file or directory, open 'doesnotexist.js'`))
+    .catch(common.oops(this)))
 
   it('should preview an empty composition', () => cli.do(`app preview data/composer/composer-source/empty.js`, this.app)
     .then(verifyTheBasicStuff('empty.js', 'composerLib'))

--- a/plugins/plugin-apache-composer/tests/data/composer/composer-source-expect-errors/openwhisk-composer-throw-err.js
+++ b/plugins/plugin-apache-composer/tests/data/composer/composer-source-expect-errors/openwhisk-composer-throw-err.js
@@ -1,0 +1,3 @@
+const composer = require('openwhisk-composer')
+
+module.exports = composer.sequence(composer.action('A', { filename: 'doesnotexist.js' }))

